services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusService.kt

```diff
@@ -1,21 +1,81 @@
 package fi.hel.haitaton.hanke.hakemus
 
+import fi.hel.haitaton.hanke.HANKEALUE_DEFAULT_NAME
+import fi.hel.haitaton.hanke.HankeEntity
 import fi.hel.haitaton.hanke.HankeNotFoundException
 import fi.hel.haitaton.hanke.HankeRepository
+import fi.hel.haitaton.hanke.HankealueService
+import fi.hel.haitaton.hanke.application.ApplicationAlreadySentException
+import fi.hel.haitaton.hanke.application.ApplicationArea
 import fi.hel.haitaton.hanke.application.ApplicationContactType
 import fi.hel.haitaton.hanke.application.ApplicationData
 import fi.hel.haitaton.hanke.application.ApplicationEntity
+import fi.hel.haitaton.hanke.application.ApplicationGeometryException
+import fi.hel.haitaton.hanke.application.ApplicationGeometryNotInsideHankeException
 import fi.hel.haitaton.hanke.application.ApplicationNotFoundException
 import fi.hel.haitaton.hanke.application.ApplicationRepository
 import fi.hel.haitaton.hanke.application.CableReportApplicationData
+import fi.hel.haitaton.hanke.domain.NewGeometriat
+import fi.hel.haitaton.hanke.domain.NewHankealue
+import fi.hel.haitaton.hanke.geometria.GeometriatDao
+import fi.hel.haitaton.hanke.logging.ApplicationLoggingService
+import fi.hel.haitaton.hanke.permissions.HankeKayttajaService
+import fi.hel.haitaton.hanke.toJsonString
+import java.time.ZonedDateTime
+import java.util.UUID
+import kotlin.reflect.KClass
+import mu.KotlinLogging
+import org.geojson.Feature
+import org.geojson.FeatureCollection
 import org.springframework.stereotype.Service
 import org.springframework.transaction.annotation.Transactional
 
+private val logger = KotlinLogging.logger {}
+
 @Service
 class HakemusService(
     private val applicationRepository: ApplicationRepository,
     private val hankeRepository: HankeRepository,
+    private val geometriatDao: GeometriatDao,
+    private val hankealueService: HankealueService,
+    private val applicationLoggingService: ApplicationLoggingService,
+    private val hankeKayttajaService: HankeKayttajaService
 ) {
+    @Transactional
+    fun createHakemus(hakemus: Hakemus, userId: String): Hakemus {
```

Comment:
Why is this added here? There are other tickets for re-implementing the hakemus creation. As-is, this method is useless and riddled with problems.

1. Production code shouldn't be added to make testing easier. When writing production code just for tests, the implementation tends to become skewed towards testing and it becomes hard to tell what code is actually needed and what is not. This method is only called from the 
2. This method would be the one we use for the `POST /hakemus` endpoint, for creating an application for an existing hanke. We agreed in https://helsinkisolutionoffice.atlassian.net/browse/HAI-2374 that it should only accept the fields that are on the first page of the application form.
3. Even if we hadn't agreed to do that, there should be a separate DTO for creating a hakemus, which would be quite close to the update DTO.
4. The internal status fields (`alluid`, `applicationIdentifier`, `allustatus`) should never be updated based on input.
5. Please add a commit message that describes the changes. Things like this should definitely be mentioned in the commit message, along with an explanation of why it was added.

Reply:
6. There are no tests for the method. Tests should in place as soon as a method is added, not added at a later point.

Reply:
You're right of course. I tried to follow what has been done in Application and Hanke Service integration tests where the Service is used to create test data. But now, since there is no create method yet in this new HankeService, I should not have put it there but instead in HakemusBuilder or HakemusFactory. The creation needs to be done inside Transaction though and I had trouble to get it done without Service using only Repositories to persist the test data. I tried add @Transactional in the test method itself but that did not seem right either. So, in the end after trying many different paths I ended up putting it into HakemuService. Which was wrong as you  have pointed out. I will refactor and put the creation of test data in test factories / builders. Maybe also without any checks as the purpose is only have the database populated with test data. Or maybe I keep the checks (for geometries and hanke users etc) so that the test data is valid. Or should I do HAI-2374 first before this and use its create method after it has been done right? As have been done in ApplicationServiceITest.

Reply:
Do you need to do something you can't do with the existing HakemusFactory / ApplicationBuilder? ApplicationBuilder returns an Application, but it uses the new customers and contacts and it's of course saved in the same DB as any Hakemus would be saved to. So the first approach should be to expand on the existing builder.

Reply:
The problem is to have a non-entity object (Hakemus) from an entity object (ApplicationEntity) without a session. This is not possible due to ApplicationEntity.yhteystiedot being lazily fetched. So, I need a place where I can persist an ApplicationEntity and return it as a full domain object. E.g. HakemusBuilder.saveWithyhteystiedot returns the entity, not a domain object. If I try to access any lazy fields (e.g. yhteystiedot) of the entity, I get an error. But, now I got it to work by doing the persisting save in HakemusFactory.save with a transaction. Maybe, when there is a proper HakemusService.create we can use that in tests?

Reply:
Do you even need the Hakemus domain object in this PR? It's only used in generating test data, which can be generated by other means.

Reply:
Hmm.. yes.. good point. Why did I create it in the first place.. was my plan at some point to use it in Service layer and smaller Request/Response dtos in Controller only.. but that was not a good plan.. either. Or maybe I thought of doing some "modulith" things and keeping entities and such only for "hakemus" package and communication with other packages (modules) would be done with the Hakemus object (and perhaps hopefully make things more clear with clear boudaries). Hmm hmm.. I'll remove it because you're right, I do not need it in this PR. But, that does not mean there is no problem of creating a pojo (HakemusUpdateRequest in this case) using an entity when there are lazy fields.

Reply:
I got rid of Hakemus and got testing to work with just repositories. Test code could use some simplyfication with e.g. extension functions.